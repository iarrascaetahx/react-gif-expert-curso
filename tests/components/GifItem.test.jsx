import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Pruebas en Gif Item', () => {

    const title = 'Tanjiro';
    const url = 'https://demon-slayer.com/tanjiro.jpg';

    test('Match con el Snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('mostrar imagen con el url y el alt indicado', () => {
        render(<GifItem title={title} url={url} />);
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe de mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
     })
})