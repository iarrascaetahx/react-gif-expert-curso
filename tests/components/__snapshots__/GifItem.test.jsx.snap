// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Gif Item Match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Tanjiro"
      src="https://demon-slayer.com/tanjiro.jpg"
    />
    <p>
      Tanjiro
    </p>
  </div>
</div>
`;
